### PR TITLE
fix: change shape of filelist list data instead of re-reading it

### PR DESCRIPTION
**Pull request by roedoejet**
this allows for any processes that take place on the filelist list data (i.e. normalization) to be preserved

fixes https://github.com/EveryVoiceTTS/EveryVoice/issues/504

### PR Goal? 

Ensures that we are outputting a set of symbols without duplicates

### Fixes? 

#504 

### Feedback sought? 

sanity. code clarity

### Priority? 

medium

### Tests added? 

tests updated

### How to test? 

You could try re-running on your dataset to see how the config changes

### Confidence? 

medium

### Version change? 

none. this isn't a breaking change. no version change needed really.

### CODE REVIEW

File: everyvoice/wizard/dataset.py [hunk 1]
```diff
@@ -684,6 +685,10 @@ def reload_filelist_data_as_dict(state):
                 state.get(StepNames.data_has_header_line_step, "yes") == "yes"
             ),
         )
+        state["filelist_data"] = []
+        for row in state["filelist_data_list"][1:]:
```
Comment (SamuelLarkin) [inline everyvoice/wizard/dataset.py:689]: What is dropped in `state["filelist_data_list"][0]`?
Why are we dropping it?

**Review by SamuelLarkin — APPROVED** [review]
LGTM

File: everyvoice/wizard/dataset.py (hunk 1 above)
Comment (wiitt) [reply everyvoice/wizard/dataset.py:689]: The first element of `state["filelist_data_list"]` is the header

**Comment by marctessier** [comment]
I ran some tests. List of characters is much shorter now without the duplicates.

ex: NEW: ( site note I manually added  ':' for moh )

```
` TEST_characters: [' ', ':' , a, e, h, i, k, l, m, n, o, r, s, t, w, y, à, á, è, é,
    ì, í, ò, ó, ě, ό, –, ’]
  TEST_phones: [d, d͡z, d͡ʒ, e, èː, é, éː, ě, f, h, i, ìː, í, íː, j, k,
    kʷ, l, m, n, n̥, o, òː, ó, óː, s, t, t͡s, ũ, ũ̀ː, ṹ, ṹː, w, z, ɑ, ɑ̀ː,
    ɑ́, ɑ́ː, ɡ, ɡʷ, ɽ, ɽ̥, ʌ̃, ʌ̃̀ː, ʌ̃́, ʌ̃́ː, ʔ, ό]`
```

Previous version: 
```
`TEST_characters: [' ', '!', '"', '''', (, ), ',', ., /, ':', ;, '?', a, á, b, e,
    é, h, i, ì, í, k, m, n, o, ó, p, r, s, t, v, w, x, y, a, à, á, b, c, d, e, è,
    é, g, h, i, ì, í, k, l, m, n, o, ò, ó, p, r, s, t, u, v, w, x, y, ' ', á, è, é,
    ì, í, ò, ó, à, á, è, é, ì, í, ò, ó, ě, ό, –, —, ’, “, ”]
  TEST_phones: [b, c, d, d͡z, d͡ʒ, e, èː, é, éː, ě, f, g, h, i, ìː, í, íː, j, k, kʷ,
    l, m, n, n̥, o, òː, ó, óː, p, s, t, t͡s, t͡ʃ, u, ũ, ũ̀ː, ṹ, ṹː, v, w, x, z, ɑ,
    ɑ̀ː, ɑ́, ɑ́ː, ɡ, ɡʷ, ɽ, ɽ̥, ʃ, ʌ̃, ʌ̃̀ː, ʌ̃́, ʌ̃́ː, ʔ, ό]` 
```


Training worked  with no issues.

File: .github/workflows/pre-commit.yml
Comment (wiitt) [inline .github/workflows/pre-commit.yml:1]: I haven't learnt much about github workflows yet, so, I didn't check this part of the code.
I wonder though if this PR is a right place to have this type of updates incorporated.

**Comment by wiitt** [comment]
Everything looks good to me except the pre-commit part in git workflows which I don't feel confident about.

**Review by wiitt — COMMENTED** [review]
Everything looks good to me except the pre-commit part in git workflows which I don't feel confident about.

File: everyvoice/wizard/dataset.py (hunk 1 above)
Comment (SamuelLarkin) [reply everyvoice/wizard/dataset.py:681]: Got it

File: .github/workflows/pre-commit.yml
Comment (roedoejet) [reply .github/workflows/pre-commit.yml:1]: no, hmm, you're right. this shouldn't be here

File: everyvoice/wizard/dataset.py [hunk 2]
```diff
@@ -684,6 +685,10 @@ def reload_filelist_data_as_dict(state):
...
+            item = {headers[i]: row[i] for i in range(len(row))}
+            state["filelist_data"].append(item)
```
[the remark this answers is not in the document]
Comment (roedoejet) [reply everyvoice/wizard/dataset.py:683]: thanks - resolving this as it's addressed here: https://github.com/EveryVoiceTTS/EveryVoice/pull/518

**Pull request merged**